From 1e9f9ce3af60680eee5558315a5be8ff3440d83c Mon Sep 17 00:00:00 2001
From: Nickey Yang <nickey.yang@rock-chips.com>
Date: Wed, 11 Apr 2018 14:40:07 +0800
Subject: [PATCH] arm: dts: rockchip: force the bootargs for rk3288 linux

Force bootargs for bring up rk3288-linux with RKIMG_BOOT.

Change-Id: I43d03e51abddeb4c29fb4c94bbd2dba774b4abe5
Signed-off-by: Nickey Yang <nickey.yang@rock-chips.com>
---
 arch/arm/boot/dts/rk3288-evb-act8846.dts          |  2 +-
 arch/arm/boot/dts/rk3288-evb-rk808-linux.dts      |  1 +
 arch/arm/boot/dts/rk3288-evb.dtsi                 |  4 ----
 arch/arm/boot/dts/rk3288-fennec.dts               |  1 +
 arch/arm/boot/dts/rk3288-firefly-reload-linux.dts |  1 +
 arch/arm/boot/dts/rk3288-firefly.dts              |  1 +
 arch/arm/boot/dts/rk3288-linux.dtsi               | 12 ++++++++++++
 arch/arm/boot/dts/rk3288-miniarm.dts              |  1 +
 arch/arm/boot/dts/rk3288-miqi.dts                 |  1 +
 arch/arm/boot/dts/rk3288-phycore-rdk.dts          |  1 +
 arch/arm/boot/dts/rk3288-popmetal.dts             |  1 +
 11 files changed, 21 insertions(+), 5 deletions(-)
 create mode 100644 arch/arm/boot/dts/rk3288-linux.dtsi

diff --git a/arch/arm/boot/dts/rk3288-evb-act8846.dts b/arch/arm/boot/dts/rk3288-evb-act8846.dts
index be3c03283b8d..2ad38486525c 100644
--- a/arch/arm/boot/dts/rk3288-evb-act8846.dts
+++ b/arch/arm/boot/dts/rk3288-evb-act8846.dts
@@ -40,7 +40,7 @@
 
 /dts-v1/;
 #include "rk3288-evb.dtsi"
-
+#include "rk3288-linux.dtsi"
 / {
 	compatible = "rockchip,rk3288-evb-act8846", "rockchip,rk3288";
 
diff --git a/arch/arm/boot/dts/rk3288-evb-rk808-linux.dts b/arch/arm/boot/dts/rk3288-evb-rk808-linux.dts
index 03d2969f010f..f32b6ba90d05 100644
--- a/arch/arm/boot/dts/rk3288-evb-rk808-linux.dts
+++ b/arch/arm/boot/dts/rk3288-evb-rk808-linux.dts
@@ -40,6 +40,7 @@
 
 /dts-v1/;
 #include "rk3288-evb.dtsi"
+#include "rk3288-linux.dtsi"
 
 / {
 	compatible = "rockchip,rk3288-evb-rk808-linux", "rockchip,rk3288";
diff --git a/arch/arm/boot/dts/rk3288-evb.dtsi b/arch/arm/boot/dts/rk3288-evb.dtsi
index 43ed2fed00f9..9a64da482781 100644
--- a/arch/arm/boot/dts/rk3288-evb.dtsi
+++ b/arch/arm/boot/dts/rk3288-evb.dtsi
@@ -45,10 +45,6 @@
 #include "rk3288.dtsi"
 
 / {
-	chosen {
-		bootargs = "earlycon=uart8250,mmio32,0xff690000 swiotlb=1";
-	};
-
 	edp_panel: edp-panel {
 		status = "disabled";
 
diff --git a/arch/arm/boot/dts/rk3288-fennec.dts b/arch/arm/boot/dts/rk3288-fennec.dts
index 5ed7528b7b2f..ff9b8e5de403 100644
--- a/arch/arm/boot/dts/rk3288-fennec.dts
+++ b/arch/arm/boot/dts/rk3288-fennec.dts
@@ -42,6 +42,7 @@
 
 #include <dt-bindings/clock/rockchip,rk808.h>
 #include "rk3288.dtsi"
+#include "rk3288-linux.dtsi"
 
 / {
 	model = "Rockchip RK3288 Fennec Board";
diff --git a/arch/arm/boot/dts/rk3288-firefly-reload-linux.dts b/arch/arm/boot/dts/rk3288-firefly-reload-linux.dts
index 2a45e6e76522..544f322d29f6 100644
--- a/arch/arm/boot/dts/rk3288-firefly-reload-linux.dts
+++ b/arch/arm/boot/dts/rk3288-firefly-reload-linux.dts
@@ -10,6 +10,7 @@
 #include <dt-bindings/display/media-bus-format.h>
 #include"rk3288-firefly.dtsi"
 #include"rk3288-rkisp1.dtsi"
+#include "rk3288-linux.dtsi"
 
 / {
 	model = "Firefly-RK3288";
diff --git a/arch/arm/boot/dts/rk3288-firefly.dts b/arch/arm/boot/dts/rk3288-firefly.dts
index c07fe92dc69f..c8a4f388a1d5 100644
--- a/arch/arm/boot/dts/rk3288-firefly.dts
+++ b/arch/arm/boot/dts/rk3288-firefly.dts
@@ -42,6 +42,7 @@
 
 /dts-v1/;
 #include "rk3288-firefly.dtsi"
+#include "rk3288-linux.dtsi"
 
 / {
 	model = "Firefly-RK3288";
diff --git a/arch/arm/boot/dts/rk3288-linux.dtsi b/arch/arm/boot/dts/rk3288-linux.dtsi
new file mode 100644
index 000000000000..535813357b8b
--- /dev/null
+++ b/arch/arm/boot/dts/rk3288-linux.dtsi
@@ -0,0 +1,12 @@
+/*
+ * Copyright (c) 2018 Fuzhou Rockchip Electronics Co., Ltd
+ *
+ * SPDX-License-Identifier: (GPL-2.0+ OR MIT).
+ */
+
+/ {
+	chosen {
+		bootargs = "earlyprintk console=tty1 console=ttyS2,115200n8 rw root=PARTUUID=614e0000-0000 rootfstype=ext4 rootwait";
+	};
+
+};
diff --git a/arch/arm/boot/dts/rk3288-miniarm.dts b/arch/arm/boot/dts/rk3288-miniarm.dts
index 2fbec41f0b23..8f3c406de0da 100644
--- a/arch/arm/boot/dts/rk3288-miniarm.dts
+++ b/arch/arm/boot/dts/rk3288-miniarm.dts
@@ -42,6 +42,7 @@
 #include <dt-bindings/clock/rockchip,rk808.h>
 #include "rk3288.dtsi"
 #include "rk3288-rkisp1.dtsi"
+#include "rk3288-linux.dtsi"
 
 / {
 	compatible = "rockchip,rk3288-miniarm", "rockchip,rk3288";
diff --git a/arch/arm/boot/dts/rk3288-miqi.dts b/arch/arm/boot/dts/rk3288-miqi.dts
index b90b0e5969ec..a2862c6a17f1 100644
--- a/arch/arm/boot/dts/rk3288-miqi.dts
+++ b/arch/arm/boot/dts/rk3288-miqi.dts
@@ -43,6 +43,7 @@
 /dts-v1/;
 #include <dt-bindings/pwm/pwm.h>
 #include "rk3288.dtsi"
+#include "rk3288-linux.dtsi"
 
 / {
 	compatible = "rockchip,rk3288-miqi", "rockchip,rk3288";
diff --git a/arch/arm/boot/dts/rk3288-phycore-rdk.dts b/arch/arm/boot/dts/rk3288-phycore-rdk.dts
index 654b4eb16725..8c771209a6bf 100644
--- a/arch/arm/boot/dts/rk3288-phycore-rdk.dts
+++ b/arch/arm/boot/dts/rk3288-phycore-rdk.dts
@@ -47,6 +47,7 @@
 #include <dt-bindings/input/input.h>
 #include <dt-bindings/leds/leds-pca9532.h>
 #include "rk3288-phycore-som.dtsi"
+#include "rk3288-linux.dtsi"
 
 / {
 	model = "Phytec RK3288 PCM-947";
diff --git a/arch/arm/boot/dts/rk3288-popmetal.dts b/arch/arm/boot/dts/rk3288-popmetal.dts
index 7a498177e8fb..1f918b223d68 100644
--- a/arch/arm/boot/dts/rk3288-popmetal.dts
+++ b/arch/arm/boot/dts/rk3288-popmetal.dts
@@ -43,6 +43,7 @@
 /dts-v1/;
 
 #include "rk3288.dtsi"
+#include "rk3288-linux.dtsi"
 
 / {
 	model = "PopMetal-RK3288";
-- 
2.35.3

